control-plane/agents/src/bin/core/controller/reconciler/volume/garbage_collector.rs

```diff
@@ -218,15 +225,20 @@ async fn disown_unused_replicas(
 
         let replica_in_target = target.as_ref().contains_replica(&replica.as_ref().uuid);
         let replica_online = matches!(context.registry().replica(&replica.as_ref().uuid).await, Ok(state) if state.online());
+        let replica_dirty = replica.as_ref().dirty();
+        let replica_owned_by_volume = replica.as_ref().owners.owned_by(volume.uuid());
+        let is_replica_owned_by_volume_target = replica.as_ref().owned_by_nexus(nexus_id);
+        let is_replica_healthy =
+            is_replica_healthy(context, &mut nexus_info, replica.as_ref(), volume.as_ref()).await?;
+
         if !replica_online
-            && !replica.as_ref().dirty()
-            && replica.as_ref().owners.owned_by(volume.uuid())
-            && !replica.as_ref().owners.owned_by_a_nexus()
+            && !replica_dirty
+            && replica_owned_by_volume
+            && !is_replica_owned_by_volume_target
```

Comment:
As discussed yesterday, you should instead check if the replica owner is itself owned by our volume.

Reply:
Done